precompile/reward_manager.go

```diff
@@ -0,0 +1,457 @@
+)
+
+// Singleton StatefulPrecompiledContract and signatures.
+var (
+	_ StatefulPrecompileConfig = &RewardManagerConfig{}
+
+	ErrCannotAllowFeeRecipients      = errors.New("non-enabled cannot call allowFeeRecipients")
+	ErrCannotAreFeeRecipientsAllowed = errors.New("non-enabled cannot call areFeeRecipientsAllowed")
+	ErrCannotCurrentRewardAddress    = errors.New("non-enabled cannot call currentRewardAddress")
+	ErrCannotDisableRewards          = errors.New("non-enabled cannot call disableRewards")
+	ErrCannotSetRewardAddress        = errors.New("non-enabled cannot call setRewardAddress")
+
+	ErrCannotEnableBothRewards = errors.New("cannot enable both fee recipients and reward address at the same time")
+	ErrEmptyRewardAddress      = errors.New("reward address cannot be empty")
+
+	RewardManagerABI        abi.ABI                     // will be initialized by init function
+	RewardManagerPrecompile StatefulPrecompiledContract // will be initialized by init function
+
+	rewardAddressStorageKey        = common.Hash{'r', 'a', 's', 'k'}
+	allowFeeRecipientsAddressValue = common.Hash{'a', 'f', 'r', 'a', 'v'}
+)
+
+type InitialRewardConfig struct {
+	AllowFeeRecipients bool           `json:"allowFeeRecipients"`
+	RewardAddress      common.Address `json:"rewardAddress,omitempty"`
+}
+
+func (i *InitialRewardConfig) Verify() error {
+	switch {
+	case i.AllowFeeRecipients && i.RewardAddress != (common.Address{}):
+		return ErrCannotEnableBothRewards
+		// shall we also check blackhole address here?
```

Comment:
I think we can just check this condition and treat the blackhole address the same as any other address. Can we remove the question here?

Reply:
removed question